convert-hf-to-gguf.py

```diff
@@ -1965,6 +1965,23 @@ def write_tensors(self):
             self.gguf_writer.add_tensor(new_name, data)
 
 
+@Model.register("CohereForCausalLM")
+class CommandR2Model(Model):
+    model_arch = gguf.MODEL_ARCH.COMMAND_R
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # max_position_embeddings = 8192 in config.json but model was actually
+        # trained on 128k context length
+        self.hparams["max_position_embeddings"] = 128*1024
```

Comment:
This is a bit of bikeshed I don't know what other's opinions on this are but I would be a bit more hesitant to hard-code the context size. I know the original team put 8192 in their `config.json` to not make users run out of VRAM for a context length most will not use, despite the model being able to handle 128k.

I think it might be a trap for future `.gguf` producers who don't expect that conversion script might overwrite context length from what it is on HF, in case the Cohere folks release new models on same architecture but different "real" context length. Maybe we can check if the embedding length is 8192 originally and only then switch that to 128*1024? Or some other way to identify this very specific Cohere model. Then I think it would be less likely that we overwrite context length to a wrong value for future models of the same arch. (Assuming future models of the same arch will exist that is).

Reply:
Yes good point. But my thinking was that when future Cohere models are released with more than 128k context length we could determine at that point a unique way to differentiate the two different model architectures.
Ideally there was another parameter that indicated the true context length of the model, perhaps we can ask Cohere to add that on HF.

Reply:
Added a PR here: https://huggingface.co/CohereForAI/c4ai-command-r-v01/discussions/24

Reply:
@acanis and others, Thank you for all your work in helping port our models to llama.cpp! Yes, we will add this parameter to the config.

Reply:
Thanks for the quick turnaround @saurabhdash . I updated the convert-hf-to-gguf.py to use the new value instead of hardcoding. Thanks for the suggestion @Noeda